From 455456a9abdf72250fe53eab2f06ca3161831f05 Mon Sep 17 00:00:00 2001
From: Robert Mader <robert.mader@collabora.com>
Date: Sun, 16 Apr 2023 11:21:49 +0200
Subject: [PATCH] xdg-shell: Always handle frame callbacks in
 popup_apply_state()

Just like we do in `toplevel_apply_state()`.

Closes https://gitlab.gnome.org/GNOME/mutter/-/issues/2752
---
 src/wayland/meta-wayland-xdg-shell.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/wayland/meta-wayland-xdg-shell.c b/src/wayland/meta-wayland-xdg-shell.c
index b0ce310d2a..45663aba15 100644
--- a/src/wayland/meta-wayland-xdg-shell.c
+++ b/src/wayland/meta-wayland-xdg-shell.c
@@ -1240,7 +1240,10 @@ meta_wayland_xdg_popup_apply_state (MetaWaylandSurfaceRole  *surface_role,
     finish_popup_setup (xdg_popup);
 
   if (!meta_wayland_surface_get_window (surface))
-    return;
+    {
+      meta_wayland_actor_surface_queue_frame_callbacks (actor_surface, pending);
+      return;
+    }
 
   if (pending->xdg_positioner)
     {
-- 
GitLab

